src/java/org/apache/cassandra/io/sstable/format/trieindex/TrieIndexSSTableReader.java

```diff
@@ -237,6 +237,11 @@ public long estimatedKeys()
 
     @Override
     protected RowIndexEntry getPosition(PartitionPosition key, Operator op, boolean updateCacheAndStats, boolean permitMatchPastLast, SSTableReadsListener listener)
+    {
+        return getPosition(key, op, updateCacheAndStats, permitMatchPastLast, listener, true, false);
+    }
+
+    private RowIndexEntry getPosition(PartitionPosition key, Operator op, boolean updateCacheAndStats, boolean permitMatchPastLast, SSTableReadsListener listener, boolean acceptPrefixMatchesForDataFile, boolean forceAssumeNoMatchForDataFile)
```

Comment:
I would prefer to extract the code for the new path to a new function. Any duplicated code would be much clearer  and lower risk than a pair of boolean parameters.